video_xblock/tests/test_backends.py

```diff
@@ -0,0 +1,380 @@
+        ([{'lang': 'some_other_lng'}], [{'lang': 'en'}, {'lang': 'ru'}, {'lang': 'uk'}])
+    )
+    @unpack
+    def test_filter_default_transcripts(self, transcripts, default):
+        """
+        Check transcripts are excluded from the list of available ones in video xblock.
+        """
+        for backend in self.backends:
+            player = self.player[backend](self.xblock)
+            default_transcripts = [{'lang': 'en'}, {'lang': 'ru'}, {'lang': 'uk'}]
+            res = player.filter_default_transcripts(default_transcripts, transcripts)
+            self.assertEqual(res, default)
+
+    @override_settings(ALL_LANGUAGES=ALL_LANGUAGES)
+    @data(
+        ('en', 'English'),
+        ('to', 'Tonga (Tonga Islands)'),
+        ('unknown', '')
+    )
+    @unpack
+    def test_get_transcript_language_parameters(self, lng_abbr, lng_name):
+        """
+        Check parameters of the transcript's language.
+        """
+        for backend in self.backends:
+            player = self.player[backend](self.xblock)
+            try:
+                res = player.get_transcript_language_parameters(lng_abbr)
+                self.assertEqual(res, (lng_abbr, lng_name))
+            except VideoXBlockException as ex:
+                self.assertIn(_('Not all the languages of transcripts fetched from video platform'), ex.message)
+
+    @data(
+        *zip(
+            backends,
+            [  # media urls
+                'https://www.youtube.com/watch?v=44zaxzFsthY',
+                'https://studio.brightcove.com/products/videocloud/media/videos/45263567468485',
+                'https://wi.st/medias/HRrr784kH8932Z',
+                'https://vimeo.com/202889234'
+            ],
+            media_ids  # expected media ids
+        )
+    )
+    @unpack
+    def test_media_id(self, backend, url, expected_media_id):
+        """
+        Check that media id is extracted from the video url.
+        """
+        player = self.player[backend](self.xblock)
+        res = player.media_id(url)
+        self.assertEqual(res, expected_media_id)
+
+    @data(*zip(
+        backends,
+        [  # expected results per backend
+            ('player_name',),
+            ['account_id', 'player_id', 'token', 'player_name'],
+            ('token', 'player_name'),
+            ('player_name',)
+        ]
+    ))
+    @unpack
+    def test_customize_xblock_fields_display(self, backend, expected_result):
+        """
+        Check backend allows to edit only permitted fields.
+        """
+        editable_fields = ['account_id', 'player_id', 'token', 'player_name']
+        player = self.player[backend]
+        res = player.customize_xblock_fields_display(editable_fields)
+        self.assertIsInstance(res, tuple)
+        self.assertEqual(res[-1], expected_result)
+
+    def apply_auth_mock(self, backend, event):
+        """
+        Save state of auth related entities before mocks are applied.
+        """
+        if backend == 'wistia':
```

Comment:
These sections should be pushed down to `WistiaAuthMock.apply_auth_mock()`, `YoutubeAuthMock.apply_auth_mock()`, &c.
Next, instead of passing `backend` here you pass `mock` and just whole section simplifies down to
`mock.apply_auth_mock()`